test/common/tracing/zipkin/zipkin_core_types_test.cc

```diff
@@ -461,6 +506,59 @@ TEST(ZipkinCoreTypesSpanTest, copyConstructor) {
   EXPECT_EQ(span.isSetTraceIdHigh(), span2.isSetTraceIdHigh());
 }
 
+TEST(ZipkinCoreTypesSpanTest, moveConstructor) {
+  Span span;
+
+  const uint64_t id = Util::generateRandom64();
+  const std::string id_hex = Hex::uint64ToHex(id);
+  span.setId(id);
+  span.setParentId(id);
+  span.setTraceId(id);
+  const int64_t timestamp =
+      std::chrono::duration_cast<std::chrono::microseconds>(
+          ProdSystemTimeSource::instance_.currentTime().time_since_epoch()).count();
+  span.setTimestamp(timestamp);
+  span.setDuration(3000LL);
+  span.setName("span_name");
+
+  Endpoint endpoint;
+  Annotation ann;
+  BinaryAnnotation bann;
+  std::vector<Zipkin::Annotation> annotations;
+
+  endpoint.setServiceName("my_service_name");
+  Network::Address::InstanceConstSharedPtr addr =
+      Network::Address::parseInternetAddressAndPort("192.168.1.2:3306");
+  endpoint.setAddress(addr);
+
+  ann.setValue(Zipkin::ZipkinCoreConstants::get().CLIENT_SEND);
+  ann.setTimestamp(timestamp);
+  ann.setEndpoint(endpoint);
+  annotations.push_back(ann);
+
+  span.setAnnotations(annotations);
+
+  Span span2(std::move(span));
+
+  EXPECT_EQ(id, span2.id());
```

Comment:
According to the test coverage reports with and without my implementation of the move constructor, this test is indeed exercising the move constructor.
Given that a "move" in the pure sense of the word does not happen when `std::move` is invoked, I don't think there is a safe/good way to determine whether or not an object was moved from. One (ugly) approach to automatically verify if the move constructor was actually called  would be to change the class implementation to set a member flag to a value in the case of a copy constructor execution, and to another value when the move constructor is exercised. I don't think we want that.